### PR TITLE
【prim】Layer norm vjp

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR types
Others

### PR changes
 Others 

### Describe
Pcard-66975
layer_norm vjp 规则

### CODE REVIEW

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h [hunk 1]
```diff
@@ -887,6 +982,35 @@ void gather_nd_grad(const Tensor& x,
   }
 }
 
+template <typename T>
+void dropout_grad(const Tensor& mask,
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: redefined?

**Review by JiabinYang — COMMENTED** [review]
one comment

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h [hunk 2]
```diff
@@ -844,6 +844,101 @@ void slice_grad(const Tensor& input,
...
+      scale_grad = nullptr;
+    }
+  }
+
+  if (x_grad) {
+    if (!scale_ptr) {
+      scale_cast =
+          full<T>(std::vector<int64_t>({1, shape_2}), 1.0, x_cast.dtype());
+    }
+    auto scale_out_grad = out_grad_cast * scale_cast;
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/composite_backward/composite_backward_api.h:925]: This is not a efficient code, optimize it in next pr maybe

**Review by JiabinYang — COMMENTED** [review]
another comment

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h (hunk 1 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/prim/api/composite_backward/composite_backward_api.h]: deleted

File: paddle/fluid/prim/api/composite_backward/composite_backward_api.h (hunk 2 above)
Comment (xiaoguoguo626807) [reply paddle/fluid/prim/api/composite_backward/composite_backward_api.h:925]: ok

**Pull request closed without merging**